kythe/typescript/indexer.ts

```diff
@@ -1533,6 +1533,28 @@ class Visitor {
     }
   }
 
+  /**
+   * Emits a reference from a "this" keyword to the type of the "this" object.
+   */
+  visitThisKeyword(keyword: ts.ThisExpression) {
+    const sym = this.host.getSymbolAtLocation(keyword);
+    if (!sym) {
+      // "this" refers to an object with no particular type, e.g.
+      //   let obj = {
+      //     foo() { this.foo(); }
+      //   };
+      return;
+    }
+    if (!sym.declarations || sym.declarations.length === 0) {
+      // "this" keyword is `globalThis`, which has no declarations.
```

Comment:
I believe this can also happen in like

```
class X { ... }
namespace X { ... }
```

I think we don't care about that though.

Reply:
There's a symbol declaration there, but it's not a type.